rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -811,6 +817,18 @@ private void processReleaseTransactions(Transaction rskTx) {
         }
     }
 
+    private void updateFederationCreationBlockHeights() {
+        if (!activations.isActive(RSKIP186)) {
+            return;
+        }
+
+        Long nextFederationCreationBlockHeight = provider.getNextFederationCreationBlockHeight();
+        if (nextFederationCreationBlockHeight != null) {
+            provider.setActiveFederationCreationBlockHeight(nextFederationCreationBlockHeight);
```

Comment:
hmm.. interesting question. @josedahlquist what is your thought?

Reply:
Yes, we should be checking if the active Federation became active.... The same should be done in the method `getActiveFederationCreationBlockHeight` as there might be a gap between the last call to the update_collections and the actual activation of the new Federation.
I mean that we should also check this in the `getActiveFederationCreationBlockHeight` method, and adjust accordingly (we cannot do it in that method only because that method will most likely not be mined).

The validation should be equivalent to what co.rsk.peg.FederationSupport#shouldFederationBeActive does:
```
    private boolean shouldFederationBeActive(Federation federation) {
        long federationAge = executionBlock.getNumber() - federation.getCreationBlockNumber();
        return federationAge >= bridgeConstants.getFederationActivationAge();
    }
```

If the new Federation should be active, then we have to replace the `activeFederationCreationBlockHeight` with `nextFederationCreationBlockHeight` and clear `nextFederationCreationBlockHeight`.

Reply:
done